tests/page/page-request-intercept.spec.ts

```diff
@@ -55,6 +56,50 @@ it('should fulfill response with empty body', async ({page, server, browserName,
   expect(await response.text()).toBe('');
 });
 
+it('should override with defaults when intercepted response not provided', async ({page, server, browserName, browserMajorVersion}) => {
+  it.skip(browserName === 'chromium' && browserMajorVersion <= 91, 'Fails in Electron that uses old Chromium');
+  server.setRoute('/empty.html', (req, res) => {
+    res.setHeader('foo', 'bar');
+    res.end('my content');
+  });
+  await page.route('**/*', async route => {
+    route.continue();
+    // @ts-expect-error
+    await route._continueToResponse({});
+    await route.fulfill({
+      status: 201,
+    });
+  });
+  const response = await page.goto(server.EMPTY_PAGE);
+  expect(response.status()).toBe(201);
+  expect(await response.text()).toBe('');
+  expect(response.headers()).toEqual({});
+});
+
+it('should fulfill with any response', async ({page, server, browserName, browserMajorVersion}) => {
+  it.skip(browserName === 'chromium' && browserMajorVersion <= 91, 'Fails in Electron that uses old Chromium');
+
+  server.setRoute('/sample', (req, res) => {
+    res.setHeader('foo', 'bar');
+    res.end('Woo-hoo');
+  });
+  const page2 = await page.context().newPage();
+  const sampleResponse = await page2.goto(`${server.PREFIX}/sample`);
```

Comment:
We cannot use `page.goto()` here to create sampleResponse as in that case `sampleResponse.body()` would throw in because `page` already started navigating to another URL. Perhaps we should bail out if we cannot retrieve response body of the response which is passed or only accept intercepted response.